src/proof/cnf_proof.h

```diff
@@ -164,6 +164,12 @@ class CnfProof {
                            std::ostream& paren,
                            ProofLetMap &letMap) = 0;
 
+  // Actually returns std::optional<std::pair<unsigned, unsigned>>
+  // First field is true if the option is filled.
+  // Detects whether a clause has x v ~x for some x
+  // If so, returns the positive occurence's idx first, then the negative's
+  static std::tuple<bool, unsigned, unsigned> detectTrivialTautology(
```

Comment:
We actually have a [`Maybe` type](https://github.com/CVC4/CVC4/blob/master/src/util/maybe.h) that you should be able to use for those purposes.

Reply:
This is fantastic! Will do!